nrf_security: Doc updates for mbed TLS 3.0.0 (#652)
nrfconnect/sdk-nrfxlib: nrf_security/doc/configuration.rst

   :local:
   :depth: 2

> The Nordic Security Module can be enabled using `PSA driver support` or with `Legacy crypto support`.

Author: Outdated proposal

PSA driver support
******************

> The Nordic Security Module with PSA driver support can be enabled by setting :kconfig:`CONFIG_NRF_SECURITY`, and additional configurations according to :ref:`nrf_security_driver_config`.

Author: Fixed, part of the next force-push

> Enabling PSA driver support expects PSA Crypto APIs to be used.

Author: Fixed, part of next force-push

> Legacy crypto support

Author: @SebastianBoe See here for definitions of legacy crypto support

*********************

> Legacy crypto mode can be enabled by setting the Kconfig variable :kconfig:CONFIG_NORDIC_SECURITY_BACKEND`, and additional configurations according to :ref:`nrf_security_legacy_config`.

Author: Fixed, part of next force-push

> Enabling legacy crypto support allows backwards compatibility for software that require usage of Mbed TLS crypto toolbox functions prefixed with `mbedtls_`.

Author: Fixed, part of next force-push

Custom Mbed TLS configuration files
***********************************

> The Kconfig options for the Nordic Security Module (nrf_security) is used to generate an Mbed TLS configuration files used during compilation.

Reviewer: ```suggestion
The Nordic Security Module (nrf_security) Kconfig options are used to generate an Mbed TLS configuration file.
```

Author: Fixed (part of next force-push)

Reviewer: Doesn't look fixed

Although not recommended, it is possible to provide a custom Mbed TLS configuration file by disabling :kconfig:`CONFIG_GENERATE_MBEDTLS_CFG_FILE`.
See :ref:`nrf_security_tls_header`.

> 

Reviewer: Are we sure CONFIG_GENERATE_MBEDTLS_CFG_FILE still works? Has anyone tested it?

Author: Confirmed to work, resolving
